src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnCache.java

```diff
@@ -157,7 +161,8 @@ public CacheHandle lookup(Spawn spawn, SpawnExecutionContext context)
         }
         // In case the remote cache returned a failed action (exit code != 0) we treat it as a
         // cache miss
-        if (result != null && result.getExitCode() == 0) {
+        if (result != null && result.getExitCode() == 0
+            && actionResultHasRequiredFiles(result, context.getMetadataProvider(), spawn)) {
```

Comment:
move everything into `actionResultHasRequiredFiles` and name it `validCacheHit(...)`?

Reply:
Done.

Reply:
Done.

Reply:
But I chose `isValidCacheHit`.